File: Data Structures & Algorithms/string-encode-and-decode/submission-2.go
type Solution struct{}

func (s *Solution) Encode(strs []string) string {
	var output string
	for _, str := range strs {
		length := strconv.Itoa(len(str))
		output += length
		output += "|"
		output += str
	}
	fmt.Println(output)
	return output
}

func (s *Solution) Decode(encoded string) []string {
	out := []string{}

	for i := 0; i < len(encoded); {
		j := i
		for encoded[j] != '|' {
			j++
		}
		length, _ := strconv.ParseInt(encoded[i:j],10, 64)
		i = j + 1
		str := encoded[i:i + int(length)]
		out = append(out, str)
		i += int(length)
	}


	return out
}

File: Data Structures & Algorithms/missing-number/submission-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        output = 0
        for i in range(len(nums)):
            output ^= i
            output ^= nums[i]
            print(f"{i=} {nums[i]=}")
        output ^= len(nums)
        
        return output